src/constants/PropTypes.js

```diff
@@ -83,6 +83,8 @@ export const TableCardPropTypes = {
     columns: PropTypes.arrayOf(
       PropTypes.shape({
         dataSourceId: PropTypes.string.isRequired,
+        /** optional width, default is no enforced max width.  for example 150px */
+        width: PropTypes.string,
```

Comment:
How are we going to handle these without styled-components? Imperative control with css custom properties?

Additionally, if we're placing these directly into styles without sanitation and we require the unit suffix, it might be a good idea to put a custom validator on it

```
width: function(props, propName, componentName) {
    if (!/(\d*)px/.test(props[propName])) {
      return new Error(
        'Invalid prop `' + propName + '` supplied to' +
        ' `' + componentName + '`. Must contain digits with `px` suffix, ex `125px`'
      );
    }
  },
```

Not 100% on that regex, you might doublecheck it 😅